src/swf/parser/templates.ts

```diff
@@ -127,10 +127,10 @@ module Shumway.SWF.Parser {
     return readSb($bytes, $stream, size) / 65536;
   }
 
-  export function readString($bytes, $stream, length) {
+  export function readString($bytes, $stream, length?) {
```

Comment:
Given that this is hot code, I'd much prefer using `-1` for "not specified" and not making the `length` argument optional.